frontend/src/libs/components/app/app.tsx

```diff
@@ -1,15 +1,48 @@
 import { Header, RouterOutlet } from '~/libs/components/components.js';
+import { DataStatus } from '~/libs/enums/data-status.enum.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+import { actions } from '~/slices/auth/auth.js';
 
+import { Loader } from '../loader/loader.js';
 import styles from './styles.module.scss';
 
 const App: React.FC = () => {
+  const { user, dataStatus } = useAppSelector(({ auth }) => ({
+    user: auth.user,
+    dataStatus: auth.dataStatus,
+  }));
+  const dispatch = useAppDispatch();
+  const hasUser = Boolean(user);
+  const hasToken = Boolean(storage.get(StorageKey.TOKEN));
+
+  if (hasUser && !hasToken) {
+    void dispatch(actions.logout());
+  }
+
+  const isLoading = !(
+    dataStatus === DataStatus.FULFILLED || dataStatus == DataStatus.REJECTED
+  );
```

Comment:
Why not just?
```
  const isLoading = dataStatus === DataStatus.PENDING;
```

Reply:
If you do it this way it will go to sign-in page before request resolves and only then it will be redirected to root.   You can clearly see sign-in page blinking in the background on page refresh.   So I thought it's better to write check like that.

Reply:
As an option we can add  const [isLoading, setLoading] = useState(false); to component and handle request like this: 

  useEffect(() => {
    setLoading(true);
    if (hasToken) {
      dispatch(actions.getCurrentUser()).then(() => setLoading(false));
    }
  }, [hasToken, dispatch]);

Reply:
It's because joined cases on reducer for dataStatuses per different actions,, actually they may be split up. But if so - then let's leave it as it is (with your implementation) and change the way how dataStatus is managed later on